src/glewlwyd.c

```diff
@@ -53,6 +55,8 @@ int main (int argc, char ** argv) {
   struct config_elements * config = o_malloc(sizeof(struct config_elements));
   int res, use_config_file = 0, use_config_env = 0;
   struct sockaddr_in bind_address;
+  pthread_t signal_thread_id;
+  static sigset_t close_signals;
```

Comment:
Why is this variable static? Is it to be invisible in the other files?

Reply:
I made it static because a pointer to it is passed to the signal thread and I wasn't sure if that thread might still be alive (for a short time) after the local variables in main() go out of scope.

We currently don't explicitly stop that thread before exiting main. Doing so would probably be more correct, but actually won't help anyway because the program exits by calling `exit()`, not by returning from main. I'm not sure if in this case it's guaranteed that other threads are cleaned up before local variables in `main()` might be overwritten. If the variable is `static` it definitely is valid until program exit.